@smdn 

Also there are build warnings for the following line. Please check if you can address the warnings or not. Thanks.
```
/// <param name="buffer">A <see cref="Span{byte}"/> to a caller-allocated buffer.</param>
```

1) Type parameter declaration must be an identifier not a type. See also error CS0081.
2) XML comment has syntactically incorrect cref attribute 'Span{byte}'

--- a/src/LibUsbDotNet/LibUsb/UsbEndpointBase.cs
+++ b/src/LibUsbDotNet/LibUsb/UsbEndpointBase.cs
@@ -98,7 +98,7 @@ public abstract class UsbEndpointBase
     /// <summary>
     /// Synchronous bulk/interrupt transfer function.
     /// </summary>
-    /// <param name="buffer">A <see cref="Span{byte}"/> to a caller-allocated buffer.</param>
+    /// <param name="buffer">A <see cref="Span{T}"/> to a caller-allocated buffer.</param>
     /// <param name="timeout">Maximum time to wait for the transfer to complete.</param>
     /// <param name="transferLength">Number of bytes actually transferred.</param>
     /// <returns>True on success.</returns>

--- a/src/LibUsbDotNet/LibUsb/UsbEndpointReader.Async.cs
+++ b/src/LibUsbDotNet/LibUsb/UsbEndpointReader.Async.cs
@@ -13,30 +13,6 @@ public partial class UsbEndpointReader
         /// <returns>
         /// Tuple of (<see cref="Error"/> error, <see cref="int"/> transferLength). error is <see cref="Error.Success"/> on success.
         /// </returns>
-        public virtual async Task<(Error error, int transferLength)> ReadAsync(byte[] buffer, int timeout) => 
-            await ReadAsync(buffer.AsMemory(0, buffer.Length), timeout).ConfigureAwait(false);
-
-        /// <summary>
-        /// Reads data asynchronously from the current <see cref="UsbEndpointReader"/>.
-        /// </summary>
-        /// <param name="buffer">The buffer to store the received data in.</param>
-        /// <param name="offset">The position in buffer to start storing the data.</param>
-        /// <param name="length">The maximum number of bytes to receive.</param>
-        /// <param name="timeout">Maximum time to wait for the transfer to complete.  If the transfer times out, the IO operation will be cancelled.</param>
-        /// <returns>
-        /// Tuple of (<see cref="Error"/> error, <see cref="int"/> transferLength). error is <see cref="Error.Success"/> on success.
-        /// </returns>
-        public virtual async Task<(Error error, int transferLength)> ReadAsync(byte[] buffer, int offset, int length, int timeout) => 
-            await TransferAsync(buffer.AsMemory(offset, length), timeout).ConfigureAwait(false);
-        
-        /// <summary>
-        /// Reads data asynchronously from the current <see cref="UsbEndpointReader"/>.
-        /// </summary>
-        /// <param name="buffer">The buffer to store the received data in.</param>
-        /// <param name="timeout">Maximum time to wait for the transfer to complete.  If the transfer times out, the IO operation will be cancelled.</param>
-        /// <returns>
-        /// Tuple of (<see cref="Error"/> error, <see cref="int"/> transferLength). error is <see cref="Error.Success"/> on success.
-        /// </returns>
         public virtual async Task<(Error error, int transferLength)> ReadAsync(Memory<byte> buffer, int timeout) => 
             await TransferAsync(buffer, timeout).ConfigureAwait(false);
         

--- a/src/LibUsbDotNet/LibUsb/UsbEndpointReader.cs
+++ b/src/LibUsbDotNet/LibUsb/UsbEndpointReader.cs
@@ -27,7 +27,7 @@ using LibUsbDotNet.Main;
 namespace LibUsbDotNet.LibUsb;
 
 /// <summary>
-/// Contains methods for retrieving data from a <see cref="EndpointType.Bulk"/> or <see cref="EndpointType.Interrupt"/> endpoint using the overloaded <see cref="Read(byte[],int,out int)"/> functions.
+/// Contains methods for retrieving data from a <see cref="EndpointType.Bulk"/> or <see cref="EndpointType.Interrupt"/> endpoint using the overloaded <see cref="Read(Span{byte},int,out int)"/> functions.
 /// </summary>
 public partial class UsbEndpointReader : UsbEndpointBase
 {
@@ -55,32 +55,6 @@ public partial class UsbEndpointReader : UsbEndpointBase
     /// The default is 4096.
     /// </remarks>
     public static int DefReadBufferSize { get; set; } = 4096;
-
-    /// <summary>
-    /// Reads data from the current <see cref="UsbEndpointReader"/>.
-    /// </summary>
-    /// <param name="buffer">The buffer to store the received data in.</param>
-    /// <param name="timeout">Maximum time to wait for the transfer to complete.  If the transfer times out, the IO operation will be cancelled.</param>
-    /// <param name="transferLength">Number of bytes actually transferred.</param>
-    /// <returns>
-    /// <see cref="Error"/>.<see cref="Error.Success"/> on success.
-    /// </returns>
-    public virtual Error Read(byte[] buffer, int timeout, out int transferLength) 
-        => Read(buffer, 0, buffer.Length, timeout, out transferLength);
-
-    /// <summary>
-    /// Reads data from the current <see cref="UsbEndpointReader"/>.
-    /// </summary>
-    /// <param name="buffer">The buffer to store the received data in.</param>
-    /// <param name="offset">The position in buffer to start storing the data.</param>
-    /// <param name="count">The maximum number of bytes to receive.</param>
-    /// <param name="timeout">Maximum time to wait for the transfer to complete.  If the transfer times out, the IO operation will be cancelled.</param>
-    /// <param name="transferLength">Number of bytes actually transferred.</param>
-    /// <returns>
-    /// <see cref="Error"/>.<see cref="Error.Success"/> on success.
-    /// </returns>
-    public virtual Error Read(byte[] buffer, int offset, int count, int timeout, out int transferLength) 
-        => Transfer(buffer.AsSpan(offset, count), timeout, out transferLength);
 
     /// <summary>
     /// Reads data from the current <see cref="UsbEndpointReader"/>.

--- a/src/LibUsbDotNet/LibUsb/UsbEndpointWriter.Async.cs
+++ b/src/LibUsbDotNet/LibUsb/UsbEndpointWriter.Async.cs
@@ -14,30 +14,6 @@ public partial class UsbEndpointWriter
     /// <returns>
     /// Tuple of (<see cref="Error"/> error, <see cref="int"/> transferLength). error is <see cref="Error.Success"/> on success.
     /// </returns>
-    public virtual async Task<(Error error, int transferLength)> WriteAsync(byte[] buffer, int timeout) => 
-        await WriteAsync(buffer, 0, buffer.Length, timeout).ConfigureAwait(false);
-
-    /// <summary>
-    /// Writes data asynchronously to the current <see cref="UsbEndpointWriter"/>.
-    /// </summary>
-    /// <param name="buffer">The buffer storing the data to write.</param>
-    /// <param name="offset">The position in buffer to start writing the data from.</param>
-    /// <param name="length">The number of bytes to write.</param>
-    /// <param name="timeout">Maximum time to wait for the transfer to complete.  If the transfer times out, the IO operation will be cancelled.</param>
-    /// <returns>
-    /// Tuple of (<see cref="Error"/> error, <see cref="int"/> transferLength). error is <see cref="Error.Success"/> on success.
-    /// </returns>
-    public virtual async Task<(Error error, int transferLength)> WriteAsync(byte[] buffer, int offset, int length, int timeout) => 
-        await TransferAsync(buffer.AsMemory(offset, length), timeout).ConfigureAwait(false);
-    
-    /// <summary>
-    /// Writes data asynchronously to the current <see cref="UsbEndpointWriter"/>.
-    /// </summary>
-    /// <param name="buffer">The buffer storing the data to write.</param>
-    /// <param name="timeout">Maximum time to wait for the transfer to complete.  If the transfer times out, the IO operation will be cancelled.</param>
-    /// <returns>
-    /// Tuple of (<see cref="Error"/> error, <see cref="int"/> transferLength). error is <see cref="Error.Success"/> on success.
-    /// </returns>
     public virtual async Task<(Error error, int transferLength)> WriteAsync(Memory<byte> buffer, int timeout) => 
         await TransferAsync(buffer, timeout).ConfigureAwait(false);
 

--- a/src/LibUsbDotNet/LibUsb/UsbEndpointWriter.cs
+++ b/src/LibUsbDotNet/LibUsb/UsbEndpointWriter.cs
@@ -25,7 +25,7 @@ using System;
 
 namespace LibUsbDotNet.LibUsb;
 
-/// <summary>Contains methods for writing data to a <see cref="EndpointType.Bulk"/> or <see cref="EndpointType.Interrupt"/> endpoint using the overloaded <see cref="Write(byte[],int,out int)"/> functions.
+/// <summary>Contains methods for writing data to a <see cref="EndpointType.Bulk"/> or <see cref="EndpointType.Interrupt"/> endpoint using the overloaded <see cref="Write(Span{byte},int,out int)"/> functions.
 /// </summary>
 public partial class UsbEndpointWriter : UsbEndpointBase
 {
@@ -33,32 +33,6 @@ public partial class UsbEndpointWriter : UsbEndpointBase
         : base(usbDevice, alternateInterfaceID, (byte)writeEndpointID, endpointType)
     {
     }
-
-    /// <summary>
-    /// Writes data to the current <see cref="UsbEndpointWriter"/>.
-    /// </summary>
-    /// <param name="buffer">The buffer storing the data to write.</param>
-    /// <param name="timeout">Maximum time to wait for the transfer to complete.  If the transfer times out, the IO operation will be cancelled.</param>
-    /// <param name="transferLength">Number of bytes actually transferred.</param>
-    /// <returns>
-    /// <see cref="Error"/>.<see cref="Error.Success"/> on success.
-    /// </returns>
-    public virtual Error Write(byte[] buffer, int timeout, out int transferLength) 
-        => Write(buffer, 0, buffer.Length, timeout, out transferLength);
-
-    /// <summary>
-    /// Writes data to the current <see cref="UsbEndpointWriter"/>.
-    /// </summary>
-    /// <param name="buffer">The buffer storing the data to write.</param>
-    /// <param name="offset">The position in buffer to start writing the data from.</param>
-    /// <param name="count">The number of bytes to write.</param>
-    /// <param name="timeout">Maximum time to wait for the transfer to complete.  If the transfer times out, the IO operation will be cancelled.</param>
-    /// <param name="transferLength">Number of bytes actually transferred.</param>
-    /// <returns>
-    /// <see cref="Error"/>.<see cref="Error.Success"/> on success.
-    /// </returns>
-    public virtual Error Write(byte[] buffer, int offset, int count, int timeout, out int transferLength) => 
-        Transfer(buffer.AsSpan(offset, count), timeout, out transferLength);
 
     /// <summary>
     /// Writes data to the current <see cref="UsbEndpointWriter"/>.

